libraries/Microsoft.Bot.Builder.Dialogs/Choices/ChoiceFactory.cs

```diff
@@ -85,88 +97,121 @@ public static Activity Inline(List<Choice> choices, string text = null, string s
             };
 
             // Format list of choices
-            var connector = string.Empty;
-            var txt = text ?? string.Empty;
-            txt += " ";
-
-            for (var index = 0; index < choices.Count; index++)
+            var separator = string.Empty;
+            var sb = new StringBuilder();
+            if (text != null)
             {
-                var choice = choices[index];
-
-                var title = choice.Action != null && choice.Action.Title != null ? choice.Action.Title : choice.Value;
+                sb.Append(text);
+            }
+            sb.Append(' ');
 
-                txt += $"{connector}";
-                if (opt.IncludeNumbers.Value)
-                {
-                    txt += "(" + (index + 1).ToString() + ") ";
-                }
-                txt += $"{title}";
-                if (index == (choices.Count - 2))
+            var list = choices.ToList();
+            for (int i = 0; i < list.Count; i++)
+            {
+                var choice = list[i];
+                if (choice is null)
                 {
-                    connector = (index == 0 ? opt.InlineOr : opt.InlineOrMore) ?? string.Empty;
+                    throw new ArgumentException("Set of choices contained a null element, which is not allowed.", nameof(choices));
                 }
-                else
+
+                string title = GetTitle(choice);
+                sb.Append(separator);
+                if (opt.IncludeNumbers.Value)
                 {
-                    connector = opt.InlineSeparator ?? string.Empty;
+                    sb.Append('(');
+                    sb.Append(i + 1);
+                    sb.Append(") ");
                 }
+                sb.Append(title);
```

Comment:
This whole block of new code is unexpected. Can we put this in a seperate, focused, PR that includes updated and/or relevant test coverage?